### PR TITLE
FE-253 Study country

### PR DESCRIPTION
**Pull request by jbianciotto**
# Description

Choose country when applying to a study, so we can show the correct typeform 

# Jira tickets

https://trialtech.atlassian.net/browse/FE-253

# Screenshots

image
image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue) :bug:
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) :bomb::boom:
- [ ] Configuration / Setup :wrench:
- [ ] Clean up / Technical debt :hammer:

## Apps affected by this change

- [x] Web
- [ ] Landing
- [ ] Management portal

## Project code modified by this change

- [X] Web
- [ ] Landing
- [ ] Management portal
- [X] Packages

# Checklist:

- [X] I have performed a self-review of my own code :innocent:
- [X] I have tested my change in both resolutions, desktop :computer: and mobile :iphone:
- [X] My code is clean and it follows the guidelines of this project :nail_care:
- [X] My changes don't generate new warnings :ok_hand:

### CODE REVIEW

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 1]
```diff
@@ -1,141 +1,211 @@
...
+import { AgeToText, getTypeformIds, parseTypeformUrl } from '../helpers';
 import { StudyApplyProps } from '../Study.types';
 import { useSearchParams } from 'next/navigation';
 import { TypeForm } from 'ui/components';
+import { useEffect, useState } from 'react';
+import { StudyApplyContentSkeleton } from './StudyApplyContentSkeleton';
 
 export const StudyApplyContent: React.FC<StudyApplyProps> = ({ study }) => {
-  const t = useTranslations('studies.studyDetail');
+  const t = useTranslations('studies.studyDetail.preview');
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:14]: unless you need to pass the translation funtion as a prop to a child component, please combine `t`, `tStudyTypes` and `tGender` into one. you can just have  `const t = useTranslations('studies.studyDetail');` and then just add the remaining path when you call the function.

Ex: `{t('type')}` -> `{t('preview.type')}`

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 2]
```diff
@@ -1,141 +1,211 @@
...
   const tActions = useTranslations('actions');
   const tStudyTypes = useTranslations('studies.studyDetail.types');
+  const tGender = useTranslations('studies.studyDetail.gender');
   const searchParams = useSearchParams();
+  const countryIdParam = searchParams.get('country_id');
+  const campaignIdParam = searchParams.get('campaign_id');
+
+  const [loading, setLoading] = useState(true);
+  const [showCountrySelection, setShowCountrySelection] = useState(false);
+  const [typeformProps, setTypeformProps] = useState<{
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:24]: Export this type to have the definition inside the `useState`

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 3]
```diff
@@ -1,141 +1,211 @@
...
+    typeformId: string;
+    label: string;
+    campaignId: string;
+  }>(null);
+
+  useEffect(() => {
+    const typeformIds = getTypeformIds(study, campaignIdParam, countryIdParam);
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:31]: I think this function is meant to return just one form. Rename it to singular. Ex: `getForm`

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 4]
```diff
@@ -1,141 +1,211 @@
...
 
-  if (!study) return null;
+    if (typeformIds) {
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:33]:
```suggestion
    if (form) {
```

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 5]
```diff
@@ -1,141 +1,211 @@
...
+        typeformId,
+        label,
+        campaignId,
+      });
+    } else {
+      if (study.forms.length > 0) {
+        const { typeformId, campaignId, label } = parseTypeformUrl(
+          study.forms[0].url_typeform,
+        );
+        setTypeformProps({
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:45]: Why are we setting typeform data before country selection?

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx (hunk 2 above)
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:24]:
```suggestion
  const [typeformData, setTypeformData] = useState<{
```

We use "Props" to define component properties type

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 6]
```diff
@@ -1,141 +1,211 @@
...
+    campaign_id: typeformProps?.campaignId,
     study_id: study.id,
-    label: label,
+    label: typeformProps?.label,
     utm_source: searchParams.get('utm_source'),
     utm_medium: searchParams.get('utm_medium'),
     utm_campaign: searchParams.get('utm_campaign'),
   };
 
+  const countryItems = study.forms.map((form) => {
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:83]: useMemo

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 7]
```diff
@@ -1,141 +1,211 @@
...
+    const typeformIds = getTypeformIds(study, null, countryId);
+    const { typeformId, campaignId, label } = typeformIds;
+    setTypeformProps({
+      typeformId,
+      label,
+      campaignId,
+    });
+  };
+
   const ageRequirements = AgeToText(
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:69]:
```suggestion
  const ageRequirements = ageToText(
```

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 8]
```diff
@@ -1,141 +1,211 @@
...
     eligibility?.maximum_age,
   );
 
-  const { typeformId, campaignId, label } = getTypeformIds(
-    study,
-    searchParams.get('campaign_id'),
-    searchParams.get('country_id'),
-  );
-
   const typeformHidden = {
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:74]: useMemo

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 9]
```diff
@@ -1,141 +1,211 @@
...
-
-        {conditions?.length > 0 && (
-          <div className="flex flex-col gap-4 py-3 border-t-1 border-gray-medium">
-            <H4
-              label={t('preview.conditions')}
-              className="!m-0 !p-0 text-base/4"
-            />
-            <div className="flex flex-col">
-              {conditions.map((condition, index) => (
+      {' '}
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:93]: Remove

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx [hunk 10]
```diff
@@ -1,141 +1,211 @@
...
+      {loading ? (
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:94]: Took this out of the return statement and before return add:

```
if (loading) {
  return <StudyApplyContentSkeleton />
}
```

Then you just return the normal JSX

File: apps/web/app/[lang]/studies/[id]/components/StudyEligibility.tsx [hunk 11]
```diff
@@ -1,12 +1,13 @@
 import { CheckIcon, H4, P2 } from 'ui/core';
 import { StudyEligibilityProps } from '../Study.types';
 import { useTranslations } from 'intl';
-import { AgeToText, GenderToText } from '../helpers';
+import { AgeToText } from '../helpers';
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyEligibility.tsx:4]:
```suggestion
import { ageToText } from '../helpers';
```

Functions starts with lowercase

File: apps/web/app/[lang]/studies/[id]/components/StudyEligibility.tsx [hunk 12]
```diff
@@ -1,12 +1,13 @@
...
 
 export const StudyEligibility = ({
   eligibility: eligibility,
 }: StudyEligibilityProps) => {
   const t = useTranslations('studies.studyDetail.studyEligibility');
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/components/StudyEligibility.tsx:9]: Same as before, combine if their not sent as a children component prop

File: packages/ui/core/radioGroup/RadioGroup.tsx [hunk 13]
```diff
@@ -5,7 +5,7 @@ import { RadioGroup as RadioGroupUI } from '@headlessui/react';
 import { twMerge } from 'tailwind-merge';
 import { RadioGroupProps } from './RadioGroup.types';
 import { getClasses } from './helper';
-
+import NextImage from 'next/image';
```
Comment (libattistella) [inline packages/ui/core/radioGroup/RadioGroup.tsx:8]: Maybe is this one the reason why the flags take too long to load. Try with a common ``

File: packages/ui/core/radioGroup/RadioGroup.tsx [hunk 14]
```diff
@@ -52,27 +53,45 @@ export const RadioGroup = ({
...
+                  {item.src ? (
+                    <RadioGroupUI.Label
+                      as={NextImage}
+                      alt={item.title}
+                      src={item.src}
+                      width={60}
+                      height={60}
+                      className={twMerge(
+                        checked
+                          ? 'ring-2 ring-primary'
```
Comment (libattistella) [inline packages/ui/core/radioGroup/RadioGroup.tsx:79]:
```suggestion
                          ? `ring-2 ring-${accent}`
```

**Review by libattistella — CHANGES_REQUESTED** [review]

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx (hunk 7 above)
Comment (libattistella) [reply apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx:69]: Already fixed in mi last [PR](https://github.com/francodimasi/uepm-front-react/pull/182/files#diff-af2a5dfa7d9acf1ffb566528f2f9e57a9805f568f084e78f8454d1833144684b)

File: apps/web/app/[lang]/studies/[id]/components/StudyEligibility.tsx (hunk 11 above)
Comment (libattistella) [reply apps/web/app/[lang]/studies/[id]/components/StudyEligibility.tsx:4]: Already fixed in mi last [PR](https://github.com/francodimasi/uepm-front-react/pull/182/files#diff-af2a5dfa7d9acf1ffb566528f2f9e57a9805f568f084e78f8454d1833144684b)

File: apps/web/app/[lang]/studies/[id]/components/StudyEligibility.tsx (hunk 12 above)
Comment (libattistella) [reply apps/web/app/[lang]/studies/[id]/components/StudyEligibility.tsx:9]: Already fixed in mi last [PR](https://github.com/francodimasi/uepm-front-react/pull/182/files#diff-af2a5dfa7d9acf1ffb566528f2f9e57a9805f568f084e78f8454d1833144684b)

**Review by libattistella — CHANGES_REQUESTED** [review]
A few things

File: apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx (hunk 5 above)
Comment (jbianciotto) [reply apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx]: To select the first country as default, but will remove it

**Review by libattistella — APPROVED** [review]

**Pull request merged**